tiledb/sm/array_schema/dimension.cc

```diff
@@ -813,23 +818,29 @@ void Dimension::relevant_ranges(
   }
 }
 
-std::vector<bool> Dimension::covered_vec(
+tdb::pmr::vector<bool> Dimension::covered_vec(
     const NDRange& ranges,
     const Range& mbr,
-    const std::vector<uint64_t>& relevant_ranges) const {
+    const tdb::pmr::vector<uint64_t>& relevant_ranges) const {
   assert(covered_vec_func_ != nullptr);
-  return covered_vec_func_(ranges, mbr, relevant_ranges);
+  auto ret = covered_vec_func_(ranges, mbr, relevant_ranges);
+  return {
+      ret.begin(),
+      ret.end(),
+      memory_tracker_->get_resource(MemoryType::DIMENSIONS)};
 }
 
 template <>
-std::vector<bool> Dimension::covered_vec<char>(
+tdb::pmr::vector<bool> Dimension::covered_vec<char>(
     const NDRange& ranges,
     const Range& mbr,
-    const std::vector<uint64_t>& relevant_ranges) {
+    const tdb::pmr::vector<uint64_t>& relevant_ranges,
+    shared_ptr<MemoryTracker> memory_tracker) {
```

Comment:
This might be a reasonable approach, but I'm also leaning towards tracking this array at all since its a binary vector so it'll be tiny.